src/sentry/integrations/github_enterprise/integration.py

```diff
@@ -263,6 +280,212 @@ def on_create_or_update_comment_error(self, api_error: ApiError, metrics_base: s
+                    "label": _("Sync Sentry Status to Github"),
+                    "help": _(
+                        "When a Sentry issue changes status, change the status of the linked ticket in Github."
+                    ),
+                    "addButtonText": _("Add Github Project"),
+                    "addDropdown": {
+                        "emptyMessage": _("All projects configured"),
+                        "noResultsMessage": _("Could not find Github project"),
+                        "items": current_repo_items,
+                        "url": reverse(
+                            "sentry-integration-github-search",
+                            args=[self.organization.slug, self.model.id],
+                        ),
+                        "searchField": "repo",
+                    },
+                    "mappedSelectors": {
+                        "on_resolve": {"choices": GitHubIssueStatus.get_choices()},
+                        "on_unresolve": {"choices": GitHubIssueStatus.get_choices()},
+                    },
+                    "columnLabels": {
+                        "on_resolve": _("When resolved"),
+                        "on_unresolve": _("When unresolved"),
+                    },
+                    "mappedColumnLabel": _("Github Project"),
+                    "formatMessageValue": False,
+                },
+            )
+
+        context = organization_service.get_organization_by_id(
+            id=self.organization_id, include_projects=False, include_teams=False
+        )
+        assert context, "organizationcontext must exist to get org"
+        organization = context.organization
+
+        has_issue_sync = features.has("organizations:integrations-issue-sync", organization)
+
+        if not has_issue_sync:
+            for field in config:
+                field["disabled"] = True
+                field["disabledReason"] = _(
+                    "Your organization does not have access to this feature"
+                )
+
+        return config
+
+    def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
+        """
+        Update the configuration field for an organization integration.
+        """
+        if not self.org_integration:
+            return
+
+        config = self.org_integration.config
+
+        # Handle status sync configuration
+        if "sync_status_forward" in data:
+            project_mappings = data.pop("sync_status_forward")
+
+            if any(
+                not mapping["on_unresolve"] or not mapping["on_resolve"]
+                for mapping in project_mappings.values()
+            ):
+                raise IntegrationError("Resolve and unresolve status are required.")
+
+            data["sync_status_forward"] = bool(project_mappings)
+
+            IntegrationExternalProject.objects.filter(
+                organization_integration_id=self.org_integration.id
+            ).delete()
```

Comment:
I just realized, shouldn't this be in a transaction along with the `create` stuff a few lines down? We'd want these project updates to be atomic whenever possible. Same with the GitHub update_configuration work.